make dst_filename optional

hmht opened the pull request:
mucrop had to be called with both a src_filename and dst_filename, if
the user wanted to replace the src with the cropped image, he had to
supply the filename twice.

By making the dst_filename optional, cropping-and-replacing a file is
made slightly easier.

Commit:
make dst_filename optional

mucrop had to be called with both a src_filename and dst_filename, if
the user wanted to replace the src with the cropped image, he had to
supply the filename twice.

By making the dst_filename optional, cropping-and-replacing a file is
made slightly easier.

c-14 commented on mucrop.c:
```diff
@@ -303,10 +303,13 @@ int main(int argc, const char *argv[])
 	int ret = 0;
 
-	if (argc != 3) {
+	if (argc != 2 && argc != 3) {
 		usage(true);
 		return EX_USAGE;
 	}
+	if (argc == 2) {
+		dst_filename = src_filename;
+	}
```
I'd probably prefer  using a switch here (just looks prettier to me)
```suggestion
	switch (argc) {
		case 2:
			dst_filename = src_filename;
		case 3:
			break;
		default:
			usage(true);
			return EX_USAGE;
	}
```

c-14 reviewed (approved):
Sorry for taking so long to respond to this, overall LGTM

I made one suggestion to the argc check which you can either apply and then I'll merge, or I can just apply it in post

Commit:
fixup! make dst_filename optional

using a switch looks better.  I changed dst so that it is intially
unassigned, instead of only reassigning when there's too few filenames.

hmht commented:
(happy new year btw!)

c-14 commented:
Merged in 9bf9e686d750bd4b59b15fb362012e470d468d52
Thanks for the changes and Happy New Year

c-14 commented:
Huh, thought I could rebase on my local copy, push and then mark this as merged but seem not
This is merged, I just ended up doing it manually
